code/parse/lua.cpp

```diff
@@ -5618,7 +5618,7 @@ ADE_FUNC(__tostring, l_Shipclass, NULL, "Ship class name", "string", "Ship class
 	if(!ade_get_args(L, "o|s", l_Shipclass.Get(&idx), &s))
 		return ade_set_error(L, "s", "");
 
-	if(idx < 0 || idx > Num_ship_classes)
+	if(idx < 0 || idx >= static_cast<int>(Ship_info.size()))
```

Comment:
I think the new version is correct but  the whole time these conditions were wrong? Weird that this never caused issues.